### PR TITLE
Extract constants for translation

### PR DESCRIPTION
**Pull request by alshopov**
Signed-off-by: Alexander Shopov <ash@kambanaria.org>

### CODE REVIEW

**Comment by alshopov** [comment]
This needs to be agreed upon before merging https://github.com/racket/string-constants/pull/40 which actually provides the constants and translations.

**Comment by alshopov** [comment]
Two more things are connected with this patch:

- we may rename two constants pkg-manager-* to install-pkg-*
- we may remove the logic in gui-pkg-manager-lib/pkg/gui/private/common.rkt for deciding which character to use as check-mark (search `send normal-control-font screen-glyph-exists?` in gui-pkg-manager-lib/pkg/gui/private/common.rkt). It is 2019 and fonts generally provide these.

File: gui-pkg-manager-lib/pkg/gui/private/by-source.rkt [hunk 1]
```diff
@@ -393,7 +393,7 @@
         [(github) sc-install-pkg-github]
         [(file-url) sc-install-pkg-file-url]
         [(dir-url) sc-install-pkg-dir-url]
-        [else (error 'type->str "unknown type ~s\n" type)]))
+        [else (error 'type->str (string-constant install-pkg-unknown) type)]))
```
Comment (rfindler) [inline gui-pkg-manager-lib/pkg/gui/private/by-source.rkt]: I think this is an internal error, so we don't need a string constant for it.

**Comment by rfindler** [comment]
Except for the one small comment I left on the diff, this looks good to me.

File: gui-pkg-manager-lib/pkg/gui/private/by-source.rkt (hunk 1 above)
Comment (alshopov) [reply gui-pkg-manager-lib/pkg/gui/private/by-source.rkt]: @rfindler - fixing this.

**Comment by alshopov** [comment]
> ...this looks good to me.
Do you have opinion on the two issues in https://github.com/racket/gui-pkg-manager/pull/4#issuecomment-527298986?

**Comment by rfindler** [comment]
I think we should leave the check-mark searching code alone unless it is causing a problem for you?

I don't think we can really rename string constants easily. We could set something up so that they are aliases (and specified outside the translation files), tho. The concern is code out there somewhere that uses the old names but that we don't know about. I agree it is unlikely, but I don't want to break that (possibly imaginary) code without a very good reason.

**Comment by alshopov** [comment]
> leave the check-mark

OK. I see similar tricks are used elsewhere and if removing it would be better to have a general solution. No, it is not causing problems for me, just a nit on my side. 

> I don't think we can really rename string constants easily. 

OK. In such a case I will rename the _newly extracted_ constant `(pkg-manager-dialog-title "Package Manager")` to `install-pkg-dialog-title` to have one less exception.

> I don't want to break 

OK, deal.

**Comment by alshopov** [comment]
In case you want to merge - this should be preceded by https://github.com/racket/string-constants/pull/40 which should be preceded itself by https://github.com/racket/string-constants/pull/38

**Comment by rfindler** [comment]
I merged  racket/string-constants#38 but that seems to have created conflicts with racket/string-constants#40. Sorry about that. If you don't get to it before me, I can probably just make a new commit in a few hours.

**Comment by alshopov** [comment]
Will fix in about an hour

**Comment by alshopov** [comment]
@rfindler: conflicts solved, https://github.com/racket/string-constants/pull/40 is ready for merge

(on a related note: https://github.com/racket/string-constants/pull/39 is also ripe for merge)

**Comment by rfindler** [comment]
Thank you!

**Pull request merged**